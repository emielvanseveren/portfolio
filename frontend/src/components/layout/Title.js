import React from 'react'
import styled from 'styled-components'

export default function Title(){
  return (
    <div>title</div>
  )
}