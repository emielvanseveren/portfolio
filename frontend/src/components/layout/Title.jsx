import React from 'react'
import styled from 'styled-components'
import size from '../../constants/size'

const Container = styled.div`
  width: 100%;
  display: flex;
  align-items: center;
  justify-content: center;
  user-select: none;
  flex-direction: column;

  h1{
    font-size: 8rem;
    text-align: center;
    color: ${({ theme }) => theme.blob};
  }
  span{
    text-align: right;
    display: block;
    font-size: 1rem;
    color: ${({ theme }) => theme.blob};
    font-weight: 600;
  }
  @media ${size.lg}{
    h1{
      font-size: 6rem;
    }
  }
  @media ${size.md}{
    span{
      text-align: center;
    }
  }
  @media ${size.xs}{
    h1{
      font-size: 4rem;
    }
  }
`

export default function Title(){
  return (
    <Container>
      <h1>Emiel Van Severen
        <span>-Developer</span>
      </h1>
    </Container>
  )
}
