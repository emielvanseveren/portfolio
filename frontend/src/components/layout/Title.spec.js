import React from 'react'
import Title from './Title'

// should add theme provider as this is created in a parent component
describe('<Title/>', () => {
  it('Snapshot', () => {
    expect(mountWithRouterAndTheme(
      <Title/>
    )).toMatchSnapshot()
  })
})
