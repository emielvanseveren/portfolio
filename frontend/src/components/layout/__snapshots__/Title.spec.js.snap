// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Title/> Snapshot 1`] = `
<Title>
  <Title__Container>
    <StyledComponent
      forwardedComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "Title__Container-sc-1l89k4e-0",
            "isStatic": false,
            "lastClassName": "gpvpQO",
            "rules": Array [
              "width:100%;display:flex;align-items:center;justify-content:center;user-select:none;flex-direction:column;h1{font-size:8rem;text-align:center;color:",
              [Function],
              ";}span{text-align:right;display:block;font-size:1rem;color:",
              [Function],
              ";font-weight:600;}",
            ],
          },
          "displayName": "Title__Container",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "Title__Container-sc-1l89k4e-0",
          "target": "div",
          "toString": [Function],
          "usesTheme": false,
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
    >
      <div
        className="Title__Container-sc-1l89k4e-0 gpvpQO"
      >
        <h1>
          Emiel Van Severen
          <span>
            -Developer
          </span>
        </h1>
      </div>
    </StyledComponent>
  </Title__Container>
</Title>
`;
